src/itwinai/torch/distributed.py

```diff
@@ -56,6 +58,28 @@ def wrapper(self: "TorchDistributedStrategy", *args, **kwargs):
     return wrapper
 
 
+def _initialize_ray() -> None:
+    """This method is used by the RayDDPStrategy and RayDeepSpeedStrategy to initialize
+    the Ray backend if it is not already initialized.
+
+        Raises:
+            EnvironmentError: If required environment variables `IP_HEAD` or `HEAD_NODE_IP`
+                are not set. These should be set from the slurm script where the ray cluster
+                is launched.
+    """
+    if not ray.is_initialized():
+        IP_HEAD = os.environ.get("IP_HEAD")
+        HEAD_NODE_IP = os.environ.get("HEAD_NODE_IP")
```

Comment:
what's the difference in Ray between these env vars? The names are a bit confusing because it seems that they are both the IP of the head node

Reply:
Good point, I have renamed IP_HEAD to HEAD_NODE_PORT to make the difference clearer.